packages/portalnetwork/src/networks/history/history.ts

```diff
@@ -70,20 +84,19 @@ export class HistoryNetwork extends BaseNetwork {
    * @returns the bytes or Blockheader if found or else undefined
    */
   public getBlockHeaderFromDB = async (
-    blockHash: Uint8Array,
+    opt: { blockHash: Uint8Array } | { blockNumber: bigint },
     asBytes = true,
-  ): Promise<Uint8Array | BlockHeader | undefined> => {
-    const contentKey = getContentKey(HistoryNetworkContentType.BlockHeader, blockHash)
-    const value = await this.retrieve(contentKey)
-    const header =
-      value !== undefined
-        ? BlockHeaderWithProof.deserialize(fromHexString(value)).header
-        : undefined
-    return header !== undefined
-      ? asBytes
-        ? header
-        : BlockHeader.fromRLPSerializedHeader(header, { setHardfork: true })
-      : undefined
+  ): Promise<undefined | (Uint8Array | BlockHeader)> => {
+    const contentKey =
+      'blockHash' in opt
+        ? getContentKey(HistoryNetworkContentType.BlockHeader, opt.blockHash)
+        : getContentKey(HistoryNetworkContentType.BlockHeaderByNumber, opt.blockNumber)
+    const value = await this.findContentLocally(fromHexString(contentKey))
```

Comment:
Note to self, really, really, figure out how we represent these things internally.  Too many conversions here.

Reply:
Ah, this is overkill, actually.  The same thing is repeated in `findContentLocally`